model_compression_toolkit/data_generation/common/optimization_utils.py

```diff
@@ -0,0 +1,460 @@
+    """
+    Handles the optimization process for all images.
+    """
+    def __init__(self,
+                 model: Any,
+                 data_gen_batch_size: int,
+                 init_dataset: Any,
+                 optimizer: Any,
+                 image_pipeline: BaseImagePipeline,
+                 activation_extractor: ActivationExtractor,
+                 image_granularity: ImageGranularity,
+                 scheduler_step_fn: Callable,
+                 scheduler: Any,
+                 normalization_mean: List[float],
+                 normalization_std: List[float],
+                 clip_images: bool,
+                 reflection: bool,
+                 initial_lr: float,
+                 eps: float = 1e-6):
+        """
+        Constructor for the AllImagesOptimizationHandler class.
+
+        Args:
+            model (Any): The framework model used for optimization.
+            data_gen_batch_size (int): The batch size for data generation.
+            init_dataset (Any): The initial dataset used for images generation.
+            optimizer (Any): The optimizer responsible for updating model parameters during optimization.
+            image_pipeline (BaseImagePipeline): The image pipeline for processing images during optimization.
+            activation_extractor (ActivationExtractor): Extractor for layer activations.
+            image_granularity (ImageGranularity): The granularity level for image dependence when generating images.
+            scheduler_step_fn (Callable): A function that defines the scheduler step behavior.
+            scheduler (Any): The scheduler responsible for adjusting the learning rate of the optimizer over time.
+            normalization_mean (List[float]): Mean values used for image normalization.
+            normalization_std (List[float]): Standard deviation values used for image normalization.
+            clip_images (bool): Flag indicating whether to clip generated images during optimization.
+            reflection (bool): Flag indicating whether to use reflection during image generation.
+            initial_lr (float): The initial learning rate used by the optimizer.
+            eps (float, optional): A small value added for numerical stability. Defaults to 1e-6.
+
+        """
+        self.model = model
+        self.image_pipeline = image_pipeline
+        self.batch_size = data_gen_batch_size
+        self.scheduler = scheduler
+        self.scheduler_step_fn = scheduler_step_fn
+        self.image_granularity = image_granularity
+        self.clip_images = clip_images
+        self.reflection = reflection
+        self.eps = eps
+        self.targets = []
+        self.initial_lr = initial_lr
+        self.normalization_mean = normalization_mean
+        self.normalization_std = normalization_std
+        self.use_all_data_stats = False
+
+        # Determine if all data statistics should be used
+        if image_granularity in [ImageGranularity.AllImages]:
```

Comment:
Why not `image_granularity == ImageGranularity.AllImages`?

or one liner: 
` self.use_all_data_stats = image_granularity == ImageGranularity.AllImages`